proto/identity/associations/association.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package xmtp.identity.associations;
+
+import "identity/associations/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/identity/associations";
+option java_package = "org.xmtp.proto.identity.associations";
+
+// The first entry of any XID log. The XID must be deterministically derivable
+// from the address and nonce.
+// The recovery address defaults to the initial associated_address unless
+// there is a subsequent ChangeRecoveryAddress in the log.
+message CreateXid {
+  string xid = 1;
+  uint32 nonce = 2;
+  Signature initial_member_signature = 3;
+}
+
+message AddAssociation {
+  enum AssociatedRole {
+    ASSOCIATED_ROLE_UNSPECIFIED = 0;
+    ASSOCIATED_ROLE_INSTALLATION = 1;
+    ASSOCIATED_ROLE_ADDRESS = 2;
+  }
+
+  string xid = 1;
+  uint64 client_timestamp_ns = 2;
+  AssociatedRole new_member_role = 3;
+  Signature existing_member_signature = 4;
+  Signature new_member_signature = 5;
+}
+
+// An entity may have been associated with multiple roles. Clients MUST remove
+// the entity from all previously associated roles.
+message RevokeAssociation {
+  string xid = 1;
+  uint64 client_timestamp_ns = 2;
+  // Either an installation or wallet public key
+  bytes revoked_public_key = 3;
```

Comment:
I don't think we can assume all entities have public keys. For example, Smart Wallets don't.